### PR TITLE
Refactor ReplAutocompletion

### LINKED ISSUE #284: Refactor ReplAutocompletion
I suggest refactoring `ReplAutocompletion` before #257.

Currently `ReplAutocompletion` implements a few functions, which should be moved to `ModuleFinder` before #257. Those Functions all deal with finding and listing module-files and should therefore be placed in `ModuleFinder`.
##### [topLevelDefinitions](https://github.com/morganey-lang/Morganey/blob/82d3165bdc4943b835d2e0f61765327de61d8215/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala#L54)

``` scala
/** 
  * Returns a sequence of all directories and .*mgn-Files at the top level of all module paths 
  *
  * Imaging having a two folders in the morganey path `modsA` and `modsB`:
  *
  * `modsA`:
  * +---list.mgn
  * +---pair.mgn
  * |
  * \---math
  *     \---arithmetic.mgn
  *
  * `modsB`:
  * +---logic.mgn
  * +---bools.mgn
  * |
  * \---numbers
  *     +--- floats.mgn
  *     \--- whole-numbers.mgn
  *
  * In that case `topLevelDefinitions` should return
  * {{{
  * Seq(list.mgn, pair.mgn, math, logic.mgn, bools.mgn, numbers)
  * }}}
  */
def topLevelDefinitions(): Seq[File] = ???
```
##### [findAllModulesIn](https://github.com/morganey-lang/Morganey/blob/82d3165bdc4943b835d2e0f61765327de61d8215/src/main/scala/me/rexim/morganey/ReplAutocompletion.scala#L58)

``` scala
/**
  * Returns a list of tuples (a, b), where 
  * a = the root directory of this module-path
  * b = the path of a certain module
  * 
  * Requirement: `b` is a sub-path of `a`
  */
def findAllModulesIn(path: String): Seq[(File, File)] = ???
```

If we move those two functions from `ReplAutocompletion` to `ModuleFinder` and generalize them for `java.io.File` and `java.net.URI` somehow, #257 should be much easier to implement.

### PR DESCRIPTION
**Pull request by rexim**
### Checklist

- [X] ~Update docs~
- [X] Add/Update Unit Tests

---

Close #284

### CODE REVIEW

File: src/main/scala/me/rexim/morganey/module/ModuleFinder.scala [hunk 1]
```diff
@@ -34,4 +37,47 @@ class ModuleFinder(val paths: List[File]) {
     val resourceUrl = classLoader.getResource(resourcePath)
     Option(resourceUrl)
   }
+
+  /**
+    * Returns a sequence of all directories and .*mgn-Files at the top level of all module paths
+    *
+    * Imaging having a two folders in the morganey path `modsA` and `modsB`:
```
Comment (ForNeVeR) [inline src/main/scala/me/rexim/morganey/module/ModuleFinder.scala]: I'm not a language expert, but shouldn't `Imaging` be `Imagine` here?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: src/main/scala/me/rexim/morganey/module/ModuleFinder.scala (hunk 1 above)
Comment (keddelzz) [reply src/main/scala/me/rexim/morganey/module/ModuleFinder.scala]: I think so, too :)

Comment (rexim) [reply src/main/scala/me/rexim/morganey/module/ModuleFinder.scala]: Fxd

**Review by keddelzz — APPROVED** [review]

**Review by ForNeVeR — APPROVED** [review]

**Comment by rexim** [comment]
@keddelzz @ForNeVeR thanks for the review!

**Pull request merged**